### ISSUE TITLE
`Tag#parse` and `Tag#to_tag_string` don't agree on output.

### ISSUE THREAD

**Issue by lindgrenj6** [RedHatInsights/topological_inventory-core#161]
Found when writing tests for the new Tagging functionality, best way to show it is here:

```
irb(main):001:0> Tag.new(:name => "testing").to_tag_string
=> "/testing"
irb(main):002:0> Tag.parse("/testing")
=> {:value=>nil, :namespace=>"testing", :name=>nil}

```

I fixed it in the Catalog implementation by changing `Tag#to_tag_string` to this:
`"/#{namespace}/#{name}".tap { |string| string << "=#{value}" if value.present? }`

Not sure which one is wrong - it was easier to fix `Tag#to_tag_string` than to mess with `Tag#parse`, but maybe `/name` is the right output.

**Comment by bdunne** [comment]
@agrare Does it make sense that `namespace` can be nil?
I thought it was required, but it looks like only `name` is currently required.

**Comment by agrare** [comment]
@bdunne yes namespace can be nil, I believe all user created tags will have a nil namespace

**Comment by bdunne** [comment]
@agrare Is there a namespaces list or something that we can use for parsing then?

**Comment by agrare** [comment]
@bdunne from the [IPP](https://docs.google.com/document/d/1SkS5QR72umMwkyHIXqiUoy2RjBWbSg46zfh0q1__mRU/edit#heading=h.ykeogbarvydq) all it says is:

> Namespaces are common across all tenants and defined in a well-known taxonomy by the implementers. Namespaces shall be used when the source of the tag is external to the platform, such as a tag attached to an AWS or OpenShift resource

Right now the only namespaces are the `source_type.name`

**Comment by lindgrenj6** [comment]
Fixed by #162
